stages/org.osbuild.fips

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/python3
+"""
+Enables the system FIPS mode.
+
+This stage calls `fips-mode-setup` to enable system FIPS mode.
+
+Notes:
+  - Requires 'chroot' in the buildroot.
+  - Runs the 'fips-mode-setup' script from the image in the chroot.
+"""
+
+
+import os.path
+import subprocess
+import sys
+
+from osbuild import api
+from osbuild.util.mnt import mount, umount
+
+SCHEMA_2 = r"""
+"options": {
+  "additionalProperties": false,
+  "description": "Enables the system FIPS mode",
+  "properties": {
+    "boot_cfg": {
+      "type": "boolean",
+      "description": "Reconfigure the boot loader"
+    }
+  }
+}
+"""
+
+
+def main(tree, options):
+
+    boot_cfg = options.get("boot_cfg", False)
+
+    for source in ("/dev", "/sys", "/proc"):
+        target = os.path.join(tree, source.lstrip("/"))
+        os.makedirs(target, exist_ok=True)
+        mount(source, target, bind=True, ro=False)
```

Comment:
This can cause issues if the stage fails before the `umount` calls below.

The `util` module defines a contextlib-based class called `MountGuard`.  The context is meant for situations like this and makes sure bind mounts (or any kind of mount really) are unmounted then the exiting the context even if an error occurs.

Check the [ostree stage](https://github.com/osbuild/osbuild/blob/73dc9fe50ea486c2fa7bc9e1909077fb8ecb428d/stages/org.osbuild.ostree#L240-L246) for an example of how it's used.

Also, are we sure we want `ro=False` for all of them?

Reply:
That piece of code was copied from the `org.osbuild.rpm` stage, I guess it should be changed also... I believe `/proc` is the only filesystem I need to mount `rw` so I modified the code to use `MountGuard` and mount the rest of the mount points in `ro` mode. Please take a look